File: backend/src/Ctor.Domain/Entities/ProjectDocument.cs
using Ctor.Domain.Common;

namespace Ctor.Domain.Entities;
public class ProjectDocument : BaseEntity
{
    public DateTime Created { get; set; }
    public long? BuildingId { get; set; }
    public Building? Building { get; set; }

    public long? DocumentId { get; set; }
    public Document? Document { get; set; }

    public long? ProjectId { get; set; }
    public Project? Project { get; set; }
}

File: backend/src/Ctor.ReportEngine/ReportEngine/src/main/kotlin/com/ctor/reportEngine/db/entities/ProjectDocument.kt
package com.ctor.reportEngine.db.entities

import java.util.Date
import javax.persistence.*

@Entity(name = "ProjectDocument")
class ProjectDocument(
        var ProjectId: Long,
        var Created: Date,
        var BuildingId: Long? = null,
        @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
        var Id: Long? = null,
        @OneToOne
        @JoinColumn(name = "DocumentId")
        var Document : Document? = null
) {

}